pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -17,125 +17,112 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
-import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 
 /**
  * Various utilities to be used in implementing equals and hashCode.
  *
+ * Keep isEqual() for primitive type for historical reason, but try not using them.
  */
 public class EqualityUtils {
   private EqualityUtils() {
   }
 
+  @Deprecated
   public static boolean isEqual(int left, int right) {
     return left == right;
   }
 
+  @Deprecated
   public static boolean isEqual(long left, long right) {
     return left == right;
   }
 
+  @Deprecated
   public static boolean isEqual(float left, float right) {
-    return Float.compare(left, right) == 0;
+    return left == right;
```

Comment:
This basically makes any object that contains NaNs unable to be equal to another, is this needed?